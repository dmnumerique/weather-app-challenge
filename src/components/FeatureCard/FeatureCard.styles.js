import styled from 'styled-components'
import theme from '../../theme/theme.styles'

export const Card = styled.div`
  padding: 0 0 ${theme.spacing.s3} 0;
  margin: 0 0 ${theme.spacing.s3} 0;
  border-bottom: 1px solid ${theme.color.gray};

  @media ${theme.device.tablet} {
    padding: 0 0 ${theme.spacing.s1} 0;
    margin: 0 0 ${theme.spacing.s4} 0;
  }
`

export const Heading = styled.h2`
  font-size: ${theme.fonts.size.h3};
  margin: 0 0 ${theme.spacing.s1} 0;
  text-transform: uppercase;

  @media ${theme.device.tablet} {
    font-size: ${theme.fonts.size.h2};
  }
`

export const Text = styled.p`
  color: ${theme.color.gray};
`

export const InfoContainer = styled.div`
  @media ${theme.device.tablet} {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
`

export const Temp = styled.p`
  font-size: ${theme.fonts.size.h4};
  font-weight: ${theme.fonts.weight.semi};

  @media ${theme.device.tablet} {
    font-size: ${theme.fonts.size.h3};
  }
`
export const CurrentWeather = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
`
