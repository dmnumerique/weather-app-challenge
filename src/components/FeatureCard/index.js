import React from 'react'
import PropTypes from 'prop-types'
import {
  Card,
  Heading,
  Text,
  InfoContainer,
  CurrentWeather,
  Temp
} from './FeatureCard.styles'
import { iconsPath } from '../../consts/config'

const FeatureCard = ({
  location,
  date,
  icon,
  temp,
  description,
  precip,
  wind,
  uv
}) => {
  const { city } = location
  const formattedDate = new Date(date).toDateString()
  return (
    <Card>
      <Heading>{city}</Heading>
      <Text>{formattedDate}</Text>
      <Text>{description}</Text>
      <InfoContainer>
        <CurrentWeather>
          <img src={`${iconsPath}${icon}.png`} alt={description} />
          <Temp>{temp}&deg;</Temp>
        </CurrentWeather>
        <div>
          <Text>Precipitation: {precip}</Text>
          <Text>Wind: {wind}</Text>
          <Text>UV: {uv}</Text>
        </div>
      </InfoContainer>
    </Card>
  )
}

FeatureCard.propTypes = {
  location: PropTypes.shape({
    city: PropTypes.string
  }).isRequired,
  date: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temp: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  precip: PropTypes.number.isRequired,
  wind: PropTypes.number.isRequired,
  uv: PropTypes.number.isRequired
}

export default FeatureCard
