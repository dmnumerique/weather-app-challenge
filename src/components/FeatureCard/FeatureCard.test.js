/* eslint-disable react/jsx-props-no-spreading */
import renderer from 'react-test-renderer'
import React from 'react'
import FeatureCard from './index'

const defaultProps = {
  location: {
    city: 'Melbourne'
  },
  date: '2021-03-17',
  icon: 'c01d',
  temp: 19,
  description: 'Windy',
  precip: 1,
  wind: 1.75,
  uv: 7.15
}

it('should render with default props', () => {
  const tree = renderer.create(<FeatureCard {...defaultProps} />).toJSON()
  expect(tree).toMatchSnapshot()
})
