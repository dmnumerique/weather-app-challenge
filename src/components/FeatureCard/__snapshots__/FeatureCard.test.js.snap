// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with default props 1`] = `
.c0 {
  padding: 0 0 24px 0;
  margin: 0 0 24px 0;
  border-bottom: 1px solid #727272;
}

.c1 {
  font-size: 32px;
  margin: 0 0 8px 0;
  text-transform: uppercase;
}

.c2 {
  color: #727272;
}

.c5 {
  font-size: 24px;
  font-weight: 600;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:768px) {
  .c0 {
    padding: 0 0 8px 0;
    margin: 0 0 32px 0;
  }
}

@media (min-width:768px) {
  .c1 {
    font-size: 40px;
  }
}

@media (min-width:768px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 32px;
  }
}

<div
  className="c0"
>
  <h2
    className="c1"
  >
    Melbourne
  </h2>
  <p
    className="c2"
  >
    Wed Mar 17 2021
  </p>
  <p
    className="c2"
  >
    Windy
  </p>
  <div
    className="c3"
  >
    <div
      className="c4"
    >
      <img
        alt="Windy"
        src="assets/icons/c01d.png"
      />
      <p
        className="c5"
      >
        19
        Â°
      </p>
    </div>
    <div>
      <p
        className="c2"
      >
        Precipitation: 
        1
      </p>
      <p
        className="c2"
      >
        Wind: 
        1.75
      </p>
      <p
        className="c2"
      >
        UV: 
        7.15
      </p>
    </div>
  </div>
</div>
`;
